lib/core/services/dialog-config-factory.ts

```diff
@@ -0,0 +1,32 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { UserPreferencesService } from '../services/user-preferences.service';
+import { DialogConfigService } from '../services/dialog-config.service';
+import { Direction } from '@angular/cdk/bidi';
+
+export function dialogConfigFactory(
+    userPreferencesService: UserPreferencesService,
+    dialogConfigService: DialogConfigService
+): any {
+    return () => {
+        return userPreferencesService
+            .select('textOrientation')
+            .subscribe((direction: Direction) => {
+                dialogConfigService.changeDirection(direction);
```

Comment:
Can the service listen to those events without this factory?

Reply:
and how would it be initialized?

Reply:
In the service constructor?

Reply:
I mean, how would the service be initiliezed. the constructor will work if we declare it for example as a dependency of a component. at this stage it won't

Reply:
ok, makes sense.